pkg/ebpftracer/c/tracee.bpf.c

```diff
@@ -2388,30 +2399,60 @@ int BPF_PROG(trace_inet_sock_set_state, struct sock *sk, int old_state, int new_
 
     program_data_t p = {};
     p.scratch_idx = 1;
+    p.ctx = skops;
     p.event = e;
-    if (!init_program_data(&p, ctx)) {
-        goto cleanup;
-    }
-
+    p.event->args_buf.offset = 0;
+    p.event->args_buf.argnum = 0;
     __builtin_memcpy(&p.event->context.task, &netctx->taskctx, sizeof(task_context_t));
 
     if (!should_trace(&p)) {
         goto cleanup;
     }
 
     tuple_t tuple = {};
-    fill_tuple(sk, &tuple);
+    fill_tuple_from_bpf_sock(sk, &tuple);
 
     save_to_submit_buf(&p.event->args_buf, (void *) &old_state, sizeof(u32), 0);
     save_to_submit_buf(&p.event->args_buf, (void *) &new_state, sizeof(u32), 1);
     save_to_submit_buf(&p.event->args_buf, &tuple, sizeof(tuple), 2);
+    if (p.event->context.task.tid == 0) {
```

Comment:
What is the reason we filter changes by thread id `0`? 🤔

Reply:
Verifier would fail because events_perf_submit contains path to init remaining task context fields and some helpers are not available for sockopts .
Changed a bit to make it more clear.